src/istgt_lu_ctl.c

```diff
@@ -721,14 +722,38 @@ istgt_uctl_cmd_desired_rf(UCTL_Ptr uctl)
 	spec = lu->lun[0].spec;
 
 	if (drf < spec->desired_replication_factor) {
-		ISTGT_ERRLOG("desired replication factor(%d) is greater than"
-		    " requested desired replication factor(%d)\n",
-		    spec->desired_replication_factor, drf);
-		istgt_uctl_snprintf(uctl, "ERR current desired replication factor"
-		    " is already greater than existing desiredRF\n");
-		goto error_return;
-	}
-	if (drf == spec->desired_replication_factor) {
+
+		// TODO: I think this check is not required(get confirmation during review process)
+		// If user want to remove scalingup replica should we need to allow this process?
+		if ( spec->desired_replication_factor != spec->replication_factor ||
+		    drf != spec->replication_factor - 1) {
+			ISTGT_ERRLOG("current desired replication factor(%d) "
+			    "and replication factor(%d) but requested desired "
+			    "replication factor(%d) are invalid changes\n",
+			    spec->desired_replication_factor,
+			    spec->replication_factor, drf);
+			istgt_uctl_snprintf(uctl, "ERR current desired "
+			    "replication factor(%d) and replication factor(%d) "
+			    "but requested desired replication factor(%d) are "
+			    "invalid changes\n", spec->desired_replication_factor,
+			    spec->replication_factor, drf);
+			goto error_return;
+		}
+		// In case of scale down read known replica list
+		known_replica_id_list = (char **) xmalloc(sizeof(char *) * drf);
+		memset(known_replica_id_list, 0, sizeof(known_replica_id_list));
```

Comment:
It should be `memset(,,sizeof(char *) * drf)`

Reply:
I will take care of memory things somehow I was in miss assumption

Reply:
handled